raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -354,35 +348,41 @@ def clear_if_finalized(
 
 def forward_transfer_pair(
     payer_transfer: LockedTransferSignedState,
-    available_routes: List[RouteState],
+    route_state: RouteState,
+    route_state_table: List[RouteState],
     channelidentifiers_to_channels: Dict,
     pseudo_random_generator: random.Random,
     block_number: BlockNumber,
 ) -> Tuple[Optional[MediationPairState], List[Event]]:
-    """ Given a payer transfer tries a new route to proceed with the mediation.
+    """ Given a payer transfer tries the given route to proceed with the mediation.
 
     Args:
         payer_transfer: The transfer received from the payer_channel.
-        available_routes: Current available routes that may be used, it's
-            assumed that the routes list is ordered from best to worst.
+        route_state: route to be tried.
+        route_state_table: list of all candidate routes
         channelidentifiers_to_channels: All the channels available for this
             transfer.
+
         pseudo_random_generator: Number generator to generate a message id.
         block_number: The current block number.
     """
     transfer_pair = None
     mediated_events: List[Event] = list()
     lock_timeout = BlockTimeout(payer_transfer.lock.expiration - block_number)
 
-    route_infos = channel.next_channel_from_routes(
-        available_routes=available_routes,
-        channelidentifiers_to_channels=channelidentifiers_to_channels,
+    payee_channel = channelidentifiers_to_channels.get(route_state.forward_channel_id)
+
+    route_states = channel.prune_route_table(
+        route_state_table=route_state_table, selected_route=route_state
+    )
+
+    is_payee_channel_usable = payee_channel is not None and channel.is_channel_usable(
+        candidate_channel_state=payee_channel,
         transfer_amount=payer_transfer.lock.amount,
         lock_timeout=lock_timeout,
     )
 
-    if route_infos:
-        payee_channel, route_state = route_infos
+    if payee_channel is not None and is_payee_channel_usable:
```

Comment:
This doesn't look right. What if one of the other routes in `route_state_table` can be used?